### PR TITLE
nsc-events-nextjs_8_486_unit-testing-01

### LINKED ISSUE #486: Unit Testing
Unit Testing

- I will be creating unit test for the admin page
- Currently there are only simple unit tests that check on rendering of the page

### PR DESCRIPTION
**Pull request by Bejarano03**
Please use the following format for the title:
nsc-events-nextjs _ unit-testing _ 486 _Unit Testing

Please fill out this description with the following:
Resolves #486 
This pull request adds more unit testing to the admin page. I figured
out to mock the nextjs link and router. 

1. Checkout my branch
2. Run npm install for any missing dependencies
3. In console run: "npm run test:coverage"

Screenshot from 2024-06-30 20-30-06

### CODE REVIEW

**Comment by Bejarano03** [comment]
Could someone point me in the right direction for the type issue I am having with the usage of children and href?

File: tests/unit/admin.test.tsx [hunk 1]
```diff
@@ -1,12 +1,48 @@
 import '@testing-library/jest-dom'
 import { render, screen } from '@testing-library/react'
 import Page from '../../app/admin/page';
+import { useRouter } from 'next/router';
+import Link from 'next/link';
+
+// Mock next/link component
+jest.mock('next/link', () => {
+    return ({ children, href }) => {
```
Comment (IsaacJrTypes) [inline tests/unit/admin.test.tsx:9]: The linter is warning us of undeclared types. 
You can add the following to fix the lint warnings:
`return ({ children, href }: {children: React.ReactNode, href: string}) => {`

File: tests/unit/admin.test.tsx [hunk 2]
```diff
@@ -1,12 +1,48 @@
...
+
     it('renders the buttons', () => {
         render(<Page />)
         const button = screen.getAllByRole('button')
         const buttonText = screen.getByText('Edit User Role')
         expect(button && buttonText).toBeInTheDocument()
     })
+
+    it('renders the Admin component', () => {
+        renderComponent();
```
Comment (IsaacJrTypes) [inline tests/unit/admin.test.tsx:32]: I see you are creating a render function that is called before each test. Another way to do the same thing is using [beforeEach](https://www.notion.so/Unit-Testing-Guide-0fa6790bf9644c99af39b613f0973006?pvs=4#cb6a7863ac24412a86bedb5c55ded114)

**Review by IsaacJrTypes — CHANGES_REQUESTED** [review]
I think you only need to update the types in line 9 but everything looks good. Great job! 
CleanShot 2024-07-08 at 14 33 25@2x

**Review by nickolasram — APPROVED** [review]
i agree with all of Isaac's feedback and, other than the changes they requested it looks good. So I'll approve for now so that you can merge this into main once you address the changes Isaac suggested.

**Review by Alzavio — APPROVED** [review]

File: tests/unit/admin.test.tsx (hunk 1 above)
Comment (Bejarano03) [reply tests/unit/admin.test.tsx]: Thank you, this helped a lot!

**Review by intisarosman1 — APPROVED** [review]
All the tests run successfully!
Screenshot 2024-07-12 191236

**Review by Asfand00 — APPROVED** [review]
Looks good, this is what I see my end:
unit-testing-nsc_events

also looks like @IsaacJrTypes changes were implemented so it should be good to merge I believe.

Sidenote: on lines 11 and 12 for the admin.test.tsx file you have two unused imports; you can probably delete those as a way of cleaning up the file

**Pull request merged**